File: 2022-2023/MyLeetCodeSolutions/JavaSolutions/Solution1502.java
//1502. Can Make Arithmetic Progression From Sequence

import java.util.Arrays;
public class Solution1502 {
    public boolean canMakeArithmeticProgression(int[] arr) {
        Arrays.sort(arr);
        int diff = arr[1] - arr[0];
        for (int i = 1; i < arr.length - 1; i++) {
            if (diff != arr[i + 1] - arr[i]) {
                return false;
            }
        }
        return true;
    }
}

File: 2022-2023/MyLeetCodeSolutions/PythonSolutions/Solution1502.py
#1502. Can Make Arithmetic Progression From Sequence

class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        arr.sort()
        difference = abs(arr[0] - arr[1])
        for i in range(1, len(arr)-1):
            temp = abs(arr[i] - arr[i+1])
            if difference != temp:
                return False
        return True
